src/data/sources/steam/Steam.vala

```diff
@@ -495,56 +507,266 @@ namespace GameHub.Data.Sources.Steam
 				root_node = root_node.get("shortcuts") as BinaryVDF.ListNode;
 			}
 
-			var game_node = new BinaryVDF.ListNode.node(root_node.nodes.size.to_string());
+			BinaryVDF.ListNode? game_node = null;
+
+			foreach(BinaryVDF.Node node in root_node.nodes.values)
+			{
+				var tmp = node as BinaryVDF.ListNode;
+				var existing_node = tmp.get("LaunchOptions") as BinaryVDF.StringNode;
+				if(existing_node.value == "--run " + game.full_id)
+				{
+					game_node = tmp;
+					break;
+				}
+			}
+
+			if(game_node == null)
+			{
+				game_node = new BinaryVDF.ListNode.node(root_node.nodes.size.to_string());
+			}
 
-			game_node.add_node(new BinaryVDF.StringNode.node("AppName", game.name));
+			game_node.add_node(new BinaryVDF.StringNode.node("appname", game.name));
```

Comment:
We need to test if intentionally not setting "appid" here makes steam use the old format as "appid". (target+name…)

If it has that effect: good, we can predict the artwork name.
If not: new appid generation is currently unknown so we can't create shortcuts and set the artwork in one go.